recipes/open62541/all/conanfile.py

```diff
@@ -0,0 +1,300 @@
+        "enable_pubsub_eth_uadp_etf": [True, False],                            # Use ETF implementation for the ETH_UADP publish
+        "enable_pubsub_mqtt": [True, False],                                    # Enable publish/subscribe with mqtt (experimental)
+        "namespace_zero": ["minimal", "reduced", "full"]                        # Completeness of the generated namespace zero (minimal/reduced/full)
+    }
+
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+        "multithread": False,
+        "tls": "mbedtls-apache",
+        "with_libwebsockets": False,
+        "enable_amalgamation": False,
+        "enable_historizing": False,
+        "enable_methodcalls": True,
+        "enable_nodemanagement": True,
+        "enable_subscriptions": True,
+        "enable_subscriptions_events": False,
+        "enable_discovery": True,
+        "enable_discovery_multicast": False,
+        "enable_micro_emb_dev_profile": False,
+        "enable_query": False,
+        "enable_immutable_nodes": False,
+        "enable_experimental_historizing": False,
+        "enable_pubsub": False,
+        "enable_pubsub_eth_uadp": False,
+        "enable_pubsub_deltaframes": False,
+        "enable_pubsub_informationmodel": False,
+        "enable_pubsub_informationmodel_methods": False,
+        "enable_pubsub_custom_publish_handling": False,
+        "enable_json_encoding": False,
+        "enable_statuscode_descriptions": True,
+        "enable_typedescription": True,
+        "enable_nodeset_compiler_descriptions": True,
+        "enable_malloc_singleton": False,
+        "msvc_force_static_crt": True,
+        "enable_discovery_semaphore": True,
+        "debug_dump_pkgs": False,
+        "enable_hardening": True,
+        "enable_subscriptions_alarms_conditions": False,
+        "enable_parsing": True,
+        "force_cpp": False,
+        "enable_da": True,
+        "enable_pubsub_eth_uadp_xdp": False,
+        "enable_pubsub_eth_uadp_etf": False,
+        "enable_pubsub_mqtt": False,
+        "namespace_zero": "reduced"
+    }
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+        if not self.options.force_cpp:
+            del self.settings.compiler.cppstd
+            del self.settings.compiler.libcxx
+
+    def configure(self):
+        self.options["libwebsockets"].with_ssl = self.options.tls
+
+        if self.options.enable_pubsub_mqtt                                      \
+          or self.options.enable_pubsub_informationmodel                        \
+          or self.options.enable_pubsub_informationmodel_methods                \
+          or self.options.enable_pubsub_custom_publish_handling                 \
+          or self.options.enable_pubsub_eth_uadp_etf:
+            self.options.enable_pubsub = True
```

Comment:
```suggestion
        if not self.options.enable_pubsub and (self.options.enable_pubsub_mqtt                                      
          or self.options.enable_pubsub_informationmodel                        \
          or self.options.enable_pubsub_informationmodel_methods                \
          or self.options.enable_pubsub_custom_publish_handling                 \
          or self.options.enable_pubsub_eth_uadp_etf):
            raise ConanInvalidConfiguration("The option 'enable_pubsub' must be True, if pubsub is required." )
```

Don't force option values, they change the package ID.